src/filters/time/time.ts

```diff
@@ -0,0 +1,80 @@
+import { matchPattern, contains } from 'utils/Helper';
+import { ParsedMatchSchema } from 'serina.schema';
+import { DateTime } from 'luxon';
+import TIME from './time.constants';
+import { parseMatches } from 'utils';
+
+export default class Time {
+    static parseText(text: string): ParsedMatchSchema[] {
+        const pattern = `(${TIME.FILLER_WORDS})?${TIME.ALL}`;
+        const matches = matchPattern(text, pattern);
+
+        if (!matches) return null;
+
+        return matches.map(match => {
+            const dateTimeObj = this.convertMatchToDateObj(match);
+            return parseMatches(text, match, dateTimeObj);
+        });
+    }
+
+    static convertMatchToDateObj(matchingText: string): Date {
+        let hour = null;
+        let minute = null;
+
+        Object.keys(TIME.SINGLE).forEach(key => {
+            const timePattern = TIME.SINGLE[key];
+            if (contains(matchingText, timePattern)) {
+                // match again without filler words
+                // then take the first and only matched string
+                const time = matchPattern(matchingText, TIME.ALL)[0];
+
+                let replaceAm;
+                let replacePm;
+                let timeSplit;
+
+                switch (timePattern) {
```

Comment:
Yeah, I'd say leave this for later. Abstracting it into a some function will be easier to read. But I'd much prefer to do that once we have done the relative time, as this might all get changed again when that comes in.